xos.h

```diff
@@ -101,36 +102,44 @@ namespace xos
     int lookahead_;
   };
 
-} // namespace xos
+  class StrAST
```

Comment:
Can probably put this in a nested `namespace ast` rather than prepending each of the nodes with `AST`.

We should also have two higher-level classes: a `Node` class that all other nodes will inherit from (this is what `parse` will also return) and an `Expr` class that represents all expressions and stuff like `StrAST` will inherit from.